client/mm/exchange_adaptor.go

```diff
@@ -1711,7 +1711,6 @@ func (u *unifiedExchangeAdaptor) withdraw(ctx context.Context, assetID uint32, a
 		return err
 	}
 
-	u.balancesMtx.Lock()
```

Comment:
If we don't lock it before the `Withdraw` call, we will have an incorrect available balance for a small amount of time, since the balance on the CEX will decrease before we add the new pending withdrawal.